drivers/client_documents_report/app/models/client_documents_report/report.rb

```diff
@@ -127,6 +128,66 @@ def overall_documents(client)
       client.client_files&.select { |cf| (cf.taggings.map(&:tag_id) & overall_document_tag_ids).present? }
     end
 
+    def additional_client_data_headers
+      @additional_client_data_headers ||= [].tap do |headers|
+        headers << 'Newest Entry Date'
+        if GrdaWarehouse::Config.cas_enabled?
+          headers << 'Active CAS Match'
+          headers << 'CAS Match Date'
+        end
+        headers << 'Newest Income from Any Source'
+        headers << 'Newest Total Monthly Income'
+        filter.chosen_secondary_cohorts.each do |cohort|
+          headers << cohort.name
+        end
+      end
+    end
+
+    def additional_client_data(client)
+      compute_additional_client_data[client.id]
+    end
+
+    private def compute_additional_client_data
+      @compute_additional_client_data ||= {}.tap do |client_data|
+        enrollments.preload(:client, enrollment: :income_benefits).find_each do |enrollment|
+          client_data[enrollment.client_id] ||= {}
+          # Make sure we get the most-recently started enrollment
+          client_data[enrollment.client_id]['Newest Entry Date'] ||= enrollment.entry_date
+          client_data[enrollment.client_id]['Newest Entry Date'] = enrollment.entry_date if enrollment.entry_date > client_data[enrollment.client_id]['Newest Entry Date']
```

Comment:
I found this a little convoluted -- I don't know if
```suggestion
          # Make sure we get the most-recently started enrollment
          client_data[enrollment.client_id]['Newest Entry Date'] = [client_data[enrollment.client_id]['Newest Entry Date'], enrollment.entry_date].compact.max
```
is clearer?

Reply:
I thinks so.  Thanks!